integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonLateDecodeRule.scala

```diff
@@ -743,7 +743,7 @@ class CarbonLateDecodeRule extends Rule[LogicalPlan] with PredicateHelper {
             attrName.contains(n)
           })
         case ar : ArrayType =>
-          attrName.contains(a.name + "[")
+          attrName.contains(a.name + "[") || ar.elementType.isInstanceOf[StructType]
```

Comment:
Don't check on the outut,
Add as below to check the child types also.
`attrName.contains(a.name + "[") || attrName.contains(a.name + ".")`